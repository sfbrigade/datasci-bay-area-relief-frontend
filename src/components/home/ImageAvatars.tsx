import React from 'react';
import { styled } from '@mui/material/styles';
import Avatar from '@mui/material/Avatar';
import Grid from '@mui/material/Grid';
import Typography from '@mui/material/Typography';
import Box from '@mui/material/Box';

import rocio from '../../assets/avatars/rocio.jpg';
import emily from '../../assets/avatars/emily.jpg';

const PREFIX = 'ImageAvatars';

const classes = {
  root: `${PREFIX}-root`,
  large: `${PREFIX}-large`
};

const StyledBox = styled(Box)((
  {
    theme
  }
) => ({
  [`& .${classes.root}`]: {
    display: 'flex',
    '& > *': {
      margin: theme.spacing(1),
    },
  },

  [`& .${classes.large}`]: {
    width: theme.spacing(15),
    height: theme.spacing(15),
  }
}));

export default function ImageAvatars() {


  return (
    <StyledBox component="section" mt={8} mb={8}>
      <Typography variant="h3">Meet the Team</Typography>
    <div className={classes.root}>
      <Grid container spacing={2}>
        <Grid item xs={6}>
          <Box display="flex"  
          flexDirection="column"
          alignItems="center"
          justifyContent="center">
        <Avatar alt="Rocio" src={rocio} className={classes.large} />
        <Typography>Rocio Ng</Typography>
        <Typography variant="body2">Project Lead</Typography>
        </Box>
        </Grid>
        <Grid item xs={6}>
        <Box display="flex"  
          flexDirection="column"
          alignItems="center"
          justifyContent="center">
        <Avatar alt="Emily" src={emily} className={classes.large} />
        <Typography>Emily Kasa</Typography>
        <Typography  variant="body2">Data Jam Lead</Typography>
        </Box>
        </Grid>
      </Grid>
    </div>
    </StyledBox>
  );
}