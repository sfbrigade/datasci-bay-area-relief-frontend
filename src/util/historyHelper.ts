import {CurrentFilters, HomeSearchFormTypes} from "../types";
import {Location} from "history";

const grabCurrentFiltersFromURLParams = (location: Location): CurrentFilters => {
  const pathFilters: HomeSearchFormTypes = {};
  if (location.search) {
    const params = new URLSearchParams(location.search);
    if (params.get("orgType")) {
      pathFilters["orgType"] = [params.get("orgType")];
    }
    if (params.get("county")) {
      pathFilters["county"] = [params.get("county")];
    }
  }
  return pathFilters;
};

export {
  grabCurrentFiltersFromURLParams
};
