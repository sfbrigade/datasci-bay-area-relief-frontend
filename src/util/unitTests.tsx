import * as ShallowRenderer from 'react-test-renderer/shallow';

export const shallowRenderToMatchSnapshot = (
    componentInstance: JSX.Element
  ) => {
    const renderer = ShallowRenderer.createRenderer();
  
    renderer.render(componentInstance);
  
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  };