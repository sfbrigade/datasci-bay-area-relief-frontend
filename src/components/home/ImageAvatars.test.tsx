import React from "react";
import {shallowRenderToMatchSnapshot} from "../../util/unitTests";

import ImageAvatars from "./ImageAvatars";
import {render, screen} from "@testing-library/react";

describe("<ImageAvatars>", () => {
  it("renders correctly", () => {
    shallowRenderToMatchSnapshot(<ImageAvatars />);
  });

  it("displays rocio name and photo and title", () => {
    render(<ImageAvatars />);

    expect(screen.getByText("Rocio Ng")).toBeVisible();
    expect(screen.getByAltText("Rocio")).toHaveAttribute("src", "rocio.jpg");
    expect(screen.getByText("Project Lead")).toBeVisible();
  });

  it("displays emilys name and photo and title", () => {
    render(<ImageAvatars />);

    expect(screen.getByText("Emily Kasa")).toBeVisible();
    expect(screen.getByAltText("Emily")).toHaveAttribute("src", "emily.jpg");
    expect(screen.getByText("Data Jam Lead")).toBeVisible();
  });

  it("displays alexs name and photo and title", () => {
    render(<ImageAvatars />);

    expect(screen.getByAltText("Alex")).toHaveAttribute("src", "alexkerr.jpg");
    expect(screen.getByText("Alex Kerr")).toBeVisible();
    expect(screen.getByText("Engineer")).toBeVisible();
  });
});
