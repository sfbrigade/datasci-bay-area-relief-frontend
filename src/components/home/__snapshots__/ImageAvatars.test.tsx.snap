// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageAvatars> renders correctly 1`] = `
<Styled(ForwardRef(Box))
  component="section"
  mb={8}
  mt={8}
>
  <ForwardRef(Typography)
    variant="h3"
  >
    Meet the Team
  </ForwardRef(Typography)>
  <div
    className="ImageAvatars-root"
  >
    <ForwardRef(Grid)
      container={true}
      spacing={2}
    >
      <ForwardRef(Grid)
        item={true}
        xs={6}
      >
        <ForwardRef(Box)
          alignItems="center"
          display="flex"
          flexDirection="column"
          justifyContent="center"
        >
          <ForwardRef(Avatar)
            alt="Rocio"
            className="ImageAvatars-large"
            src="rocio.jpg"
          />
          <ForwardRef(Typography)>
            Rocio Ng
          </ForwardRef(Typography)>
          <ForwardRef(Typography)
            variant="body2"
          >
            Project Lead
          </ForwardRef(Typography)>
        </ForwardRef(Box)>
      </ForwardRef(Grid)>
      <ForwardRef(Grid)
        item={true}
        xs={6}
      >
        <ForwardRef(Box)
          alignItems="center"
          display="flex"
          flexDirection="column"
          justifyContent="center"
        >
          <ForwardRef(Avatar)
            alt="Emily"
            className="ImageAvatars-large"
            src="emily.jpg"
          />
          <ForwardRef(Typography)>
            Emily Kasa
          </ForwardRef(Typography)>
          <ForwardRef(Typography)
            variant="body2"
          >
            Data Jam Lead
          </ForwardRef(Typography)>
        </ForwardRef(Box)>
      </ForwardRef(Grid)>
      <ForwardRef(Grid)
        item={true}
        xs={6}
      >
        <ForwardRef(Box)
          alignItems="center"
          display="flex"
          flexDirection="column"
          justifyContent="center"
        >
          <ForwardRef(Avatar)
            alt="Alex"
            className="ImageAvatars-large"
            src="alexkerr.jpg"
          />
          <ForwardRef(Typography)>
            Alex Kerr
          </ForwardRef(Typography)>
          <ForwardRef(Typography)
            variant="body2"
          >
            Engineer
          </ForwardRef(Typography)>
        </ForwardRef(Box)>
      </ForwardRef(Grid)>
    </ForwardRef(Grid)>
  </div>
</Styled(ForwardRef(Box))>
`;
